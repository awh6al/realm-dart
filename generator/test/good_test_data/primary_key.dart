import 'package:realm_common/realm_common.dart';

//part 'nullable_primary_key.g.dart';

@RealmModel()
class _NullableIntPK {
  @PrimaryKey()
  int? id;
}

@RealmModel()
class _NullableStringPK {
  @PrimaryKey()
  String? id;
}

@RealmModel()
class _IntPK {
  @PrimaryKey()
  late int id;
}

@RealmModel()
class _StringPK {
  @PrimaryKey()
  late String id;
}
