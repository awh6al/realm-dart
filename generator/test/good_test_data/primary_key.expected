// **************************************************************************
// RealmObjectGenerator
// **************************************************************************

class NullableIntPK extends _NullableIntPK with RealmEntity, RealmObject {
  NullableIntPK(
    int? id,
  ) {
    RealmObject.set(this, 'id', id);
  }

  NullableIntPK._();

  @override
  int? get id => RealmObject.get<int>(this, 'id') as int?;
  @override
  set id(int? value) => throw RealmUnsupportedSetError();

  @override
  Stream<RealmObjectChanges<NullableIntPK>> get changes =>
      RealmObject.getChanges<NullableIntPK>(this);

  static SchemaObject get schema => _schema ??= _initSchema();
  static SchemaObject? _schema;
  static SchemaObject _initSchema() {
    RealmObject.registerFactory(NullableIntPK._);
    return const SchemaObject(NullableIntPK, [
      SchemaProperty('id', RealmPropertyType.int,
          optional: true, primaryKey: true),
    ]);
  }
}

class NullableStringPK extends _NullableStringPK with RealmEntity, RealmObject {
  NullableStringPK(
    String? id,
  ) {
    RealmObject.set(this, 'id', id);
  }

  NullableStringPK._();

  @override
  String? get id => RealmObject.get<String>(this, 'id') as String?;
  @override
  set id(String? value) => throw RealmUnsupportedSetError();

  @override
  Stream<RealmObjectChanges<NullableStringPK>> get changes =>
      RealmObject.getChanges<NullableStringPK>(this);

  static SchemaObject get schema => _schema ??= _initSchema();
  static SchemaObject? _schema;
  static SchemaObject _initSchema() {
    RealmObject.registerFactory(NullableStringPK._);
    return const SchemaObject(NullableStringPK, [
      SchemaProperty('id', RealmPropertyType.string,
          optional: true, primaryKey: true),
    ]);
  }
}

class IntPK extends _IntPK with RealmEntity, RealmObject {
  IntPK(
    int id,
  ) {
    RealmObject.set(this, 'id', id);
  }

  IntPK._();

  @override
  int get id => RealmObject.get<int>(this, 'id') as int;
  @override
  set id(int value) => throw RealmUnsupportedSetError();

  @override
  Stream<RealmObjectChanges<IntPK>> get changes =>
      RealmObject.getChanges<IntPK>(this);

  static SchemaObject get schema => _schema ??= _initSchema();
  static SchemaObject? _schema;
  static SchemaObject _initSchema() {
    RealmObject.registerFactory(IntPK._);
    return const SchemaObject(IntPK, [
      SchemaProperty('id', RealmPropertyType.int, primaryKey: true),
    ]);
  }
}

class StringPK extends _StringPK with RealmEntity, RealmObject {
  StringPK(
    String id,
  ) {
    RealmObject.set(this, 'id', id);
  }

  StringPK._();

  @override
  String get id => RealmObject.get<String>(this, 'id') as String;
  @override
  set id(String value) => throw RealmUnsupportedSetError();

  @override
  Stream<RealmObjectChanges<StringPK>> get changes =>
      RealmObject.getChanges<StringPK>(this);

  static SchemaObject get schema => _schema ??= _initSchema();
  static SchemaObject? _schema;
  static SchemaObject _initSchema() {
    RealmObject.registerFactory(StringPK._);
    return const SchemaObject(StringPK, [
      SchemaProperty('id', RealmPropertyType.string, primaryKey: true),
    ]);
  }
}